Assets/_BForBoss/_Core/Scripts/Character/PlayerDashBehaviour.cs

```diff
@@ -55,30 +62,36 @@ public void HandleInput()
             }
         }
         
-        public void OnDashing()
+        public void DashDuringMovement()
         {
-            if (!_isDashing)
+            if (_isDashing)
             {
-                return;
-            }
+                if (_baseCharacter.IsFalling())
+                {
+                    _baseCharacter.SetVelocity(GetDashVelocity());
+                }
 
-            if (_baseCharacter.IsFalling())
-            {
-                _baseCharacter.SetVelocity(GetDashVelocity());
-            }
-
-            _dashElapsedTime += Time.deltaTime;
-            if (_dashElapsedTime > _dashDuration)
-            {
-                StopDashing();
+                _dashElapsedTime += Time.deltaTime;
+                if (_dashElapsedTime > _dashDuration)
+                {
+                    StopDashing();
+                }
             }
         }
         
-        public void OnDash(bool isDashing)
+        public void OnDash(bool isDashPressed)
```

Comment:
I combined some methods. 

Having overly simple methods is a bit of a code smell, and there were too many methods with the names of 
OnDash, OnDashing, Dash ....it get's too confusing. 

Changed some names and removed some oversimplified methods.